File: selected_coding_interview/codes/java/lc_50_powx_n/lc_50_powx_n.java
/*
* File: lc_50_powx_n.java
* Created Time: 2025-12-30
* Author: krahets
*/

package lc_50_powx_n;

import include.*;
import java.util.*;

// ===== Solution Code =====
    class Solution {
        public double myPow(double x, int n) {
            if(x == 0.0f) return 0.0d;
            long b = n;
            double res = 1.0;
            if(b < 0) {
                x = 1 / x;
                b = -b;
            }
            while(b > 0) {
                if((b & 1) == 1) res *= x;
                x *= x;
                b >>= 1;
            }
            return res;
        }
    }

public class lc_50_powx_n {
    public static void main(String[] args) {
        // ======= Test Case =======
        // Test case 1
        // TODO: Add appropriate test case

        // ====== Driver Code ======
        Solution slt = new Solution();
        // result = slt.myPow(...)
        // print(result)

    }
}

File: selected_coding_interview/codes/python/lc_50_powx_n.py
"""
File: lc_50_powx_n.py
Created Time: 2025-12-30
Author: krahets
"""

from include import *


# ===== Solution Code =====
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if x == 0.0: return 0.0
        res = 1
        if n < 0: x, n = 1 / x, -n
        while n:
            if n & 1: res *= x
            x *= x
            n >>= 1
        return res


# ======= Test Case =======
# Test case 1
# TODO: Add appropriate test case

# ====== Driver Code ======
slt = Solution()
# result = slt.myPow(...)
# print(result)
